src/main/java/ai/chalk/internal/arrow/FeatherProcessor.java

```diff
@@ -376,13 +393,19 @@ public static Table convertBytesToTable(byte[] bytes) throws Exception {
         if (maybeSingleBatchTable != null) {
             return maybeSingleBatchTable;
         }
-
-        SeekableReadChannel seekableReadChannel = new SeekableReadChannel(new ByteArrayReadableSeekableByteChannel(bytes));
-        ArrowFileReader arrowFileReader = new ArrowFileReader(seekableReadChannel, new RootAllocator(Long.MAX_VALUE), new CommonsCompressionFactory());
-
         try (
+            SeekableReadChannel seekableReadChannel = new SeekableReadChannel(
+                new ByteArrayReadableSeekableByteChannel(bytes)
+            );
+            ArrowFileReader arrowFileReader = new ArrowFileReader(
+                seekableReadChannel,
+                new RootAllocator(Long.MAX_VALUE),
```

Comment:
this allocator is leaked, all autoclosables should be defined separately to ensure they get closed

Reply:
addressing in https://github.com/chalk-ai/chalk-java/pull/104